python/ray/train/batch_predictor.py

```diff
@@ -222,13 +281,19 @@ def __call__(self, batch):
                 # Set the in-predictor preprocessing to a no-op when using a separate
                 # GPU stage. Otherwise, the preprocessing will be applied twice.
                 override_prep = BatchMapper(lambda x: x)
+                # preprocessor.transform will break for DatasetPipeline due to
+                # missing _dataset_format()
                 batch_fn = preprocessor._transform_batch
-                data = data.map_batches(batch_fn, batch_format="pandas")
+                data = data.map_batches(
+                    batch_fn, batch_format=predict_stage_batch_format
+                )
 
         prediction_results = data.map_batches(
             ScoringWrapper,
             compute=compute,
-            batch_format="pandas",
+            batch_format=preprocessor_batch_format
+            if self.get_preprocessor()
```

Comment:
Nit: Should be a `None` check rather than a truthy check.
```suggestion
            if self.get_preprocessor() is not None
```